### ISSUE TITLE
Possible bugs in simclr_lin.py

### ISSUE THREAD

**Issue by MrChenFeng** [wangxin0716/ssl#1]
Hi, I guess here in simclr_lin.py:

```
model.enc.requires_grad = False

parameters = [param for param in model.parameters() if param.requires_grad is True]  # trainable parameters.
```

I guess the right one is:
```   
for param in model.enc.parameters():

        param.requires_grad = False

parameters = [param for param in model.parameters() if param.requires_grad is True]  # trainable parameters.
```
I think in your code, actually you finetune the whole network thus you get much better performance. For what I tested, resnet18 can only achieve about 88% accuracy.

**Comment by p3i0t** [comment]
Could you please elaborate on the difference between: 
`model.enc.requires_grad = False` 
and 
 `for param in model.enc.parameters():
        param.requires_grad = False`?

**Comment by MrChenFeng** [comment]
The first assignment seems to be useless, you could check the parameter length after these two cmds.

在 2020-07-10 18:50:17，wangxin0716 <notifications@github.com> 写道：


Could you please tell the difference between:
model.enc.requires_grad = False
and
for param in model.enc.parameters(): param.requires_grad = False?

**Comment by MrChenFeng** [comment]
Hi, 
I have tried many different hyper-parameter settings, still can't reproduce the original paper result which reported cifar10 94% result, how about your experiments before?  My best result is about 89.5%.
Thanks ~

**Comment by Holim0711** [comment]
https://github.com/wangxin0716/SimCLR-CIFAR10/blob/2f449c2e39666a5c3439859347e3f1aced67b17d/simclr_lin.py#L118

The encoder seems not frozen.
If I'm not wrong, this line should be `model.enc.requires_grad_(False)`

**Comment by kfallah** [comment]
Hi @MrChenFeng, I am getting a similar result as you (a maximum linear probe accuracy of 89.5%). I am wondering if you ever resolved this issue/found a way to increase the accuracy to the reported metrics?

I noticed two other small issues in this code. The scheduler uses the initial LR and num_epochs from SimCLR training, instead of the fine-tuning one. This causes the LR to decay a lot slower.
